core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -74,6 +75,14 @@ public static MetricsConfig fromProperties(Map<String, String> props) {
     return spec;
   }
 
+  public void validateReferencedColumns(Schema schema) {
+    for (String column : columnModes.keySet()) {
+      ValidationException.check(
+          schema.findField(column) != null,
+          "Could not find column %s from metrics in schema %s", column, schema);
```

Comment:
Just thinking out loud: will be clear for an inexperienced user what is wrong?
Is `from metrics` descriptive enough? Should we make it `from metrics config`?
Or make it as `Invalid metrics config: could not find column %s in schema %s`?

Reply:
That seems like a fair criticism. What about adding the bad property name and noting that it is a table property? That would help.